services/tracing/tracing/infra/store_psql.go

```diff
@@ -12,19 +20,124 @@ func NewStorePSQL(db *sqlx.DB) *stepStore {
 }
 
 func (s *stepStore) Insert(step tracing.Step) error {
-	if _, err := s.db.Exec(
-		`INSERT INTO "steps" ("tracing_id", "step_index", "steps_remaining", "start_time", "error") VALUES ($1, $2, $3, $4, $5)`,
-		step.TracingID,
-		step.StepIndex,
-		step.StepsRemaining,
-		step.StartTime,
-		step.Error); err != nil {
+	q := sq.Insert("steps").
+		Columns("tracing_id", "step_index", "steps_remaining", "start_time", "error", "device_id").
+		Values(step.TracingID, step.StepIndex, step.StepsRemaining, step.StartTime, step.Error, step.DeviceId)
+	_, err := q.PlaceholderFormat(sq.Dollar).RunWith(s.db).Exec()
+	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+type TraceQueryPage struct {
+	StartTime time.Time `pagination:"archive.archived_at,DESC"`
+	TracingID uuid.UUID `pagination:"archive.tracing_id,DESC"`
+}
+
+// TODO: this is not a maintainable solution, the second query might receive a thousand values in the 'IN' clause
```

Comment:
Not relevant for this function correct?

Reply:
correct, however I think this is an important note so I moved it to the application layer